scipy/stats/_resampling.py

```diff
@@ -658,59 +671,65 @@ def _monte_carlo_test_iv(sample, rvs, statistic, vectorized, n_resamples,
     if alternative not in alternatives:
         raise ValueError(f"`alternative` must be in {alternatives}")
 
-    return (sample, rvs, statistic_vectorized, vectorized, n_resamples_int,
+    return (data_iv, rvs, statistic_vectorized, vectorized, n_resamples_int,
             batch_iv, alternative, axis_int)
 
 
 fields = ['statistic', 'pvalue', 'null_distribution']
 MonteCarloTestResult = make_dataclass("MonteCarloTestResult", fields)
 
 
-def monte_carlo_test(sample, rvs, statistic, *, vectorized=None,
+@_rename_parameter('sample', 'data')
+def monte_carlo_test(data, rvs, statistic, *, vectorized=None,
                      n_resamples=9999, batch=None, alternative="two-sided",
                      axis=0):
-    r"""
-    Monte Carlo test that a sample is drawn from a given distribution.
+    r"""Monte Carlo test that data is drawn from given distributions.
 
-    The null hypothesis is that the provided `sample` was drawn at random from
-    the distribution for which `rvs` generates random variates. The value of
-    the `statistic` for the given sample is compared against a Monte Carlo null
-    distribution: the value of the statistic for each of `n_resamples`
-    samples generated by `rvs`. This gives the p-value, the probability of
-    observing such an extreme value of the test statistic under the null
-    hypothesis.
+    The null hypothesis is that the provided `data` was drawn at random from
+    the distributions for which the elements of `rvs` generates random
+    variates. The value of `statistic` for the given `data` is compared against
+    a Monte Carlo null distribution: the value of the statistic for each of
+    `n_resamples` sets of samples generated by `rvs`. This gives the p-value,
+    the probability of observing such an extreme value of the test statistic
+    under the null hypothesis.
```

Comment:
I think this is correct but seems a little terse given the complexity added by this PR. The original paragraph is easy to understand, but I think I would have trouble understanding the new paragraph right away if I wasn't aware of the old one and/or not already aware of the behavior of this function. What do you think of adding a little more clarifying info like below?

```suggestion
    `data` contains a sample or a sequence of one or more samples. `rvs`
    specifies the null distribution(s) for the sample(s) in `data`. The null
    hypothesis is that the provided `data` was drawn at random from the
    distribution(s) for which `rvs` (or the elements of `rvs`) generate random
    variates. The value of `statistic` for the given `data` is compared against
    a Monte Carlo null distribution: the value of the statistic for each of
    `n_resamples` sets of samples generated by `rvs`. This gives the p-value,
    the probability of observing such an extreme value of the test statistic
    under the null hypothesis.
```

Reply:
To avoid the redundancy of the second and third sentence:
```suggestion
    `data` contains a sample or a sequence of one or more samples. `rvs`
    specifies the distribution(s) for the sample(s) in `data` under the null
    hypothesis. The value of `statistic` for the given `data` is compared against
    a Monte Carlo null distribution: the value of the statistic for each of
    `n_resamples` sets of samples generated by `rvs`. This gives the p-value,
    the probability of observing such an extreme value of the test statistic
    under the null hypothesis.
```

Reply:
Looks good to me.